packages/mask/src/plugins/CryptoartAI/SNSAdaptor/CheckoutDialog.tsx

```diff
@@ -0,0 +1,163 @@
+            justifyContent: 'space-between',
+            marginBottom: theme.spacing(0.5),
+            '&:last-child': {
+                marginBottom: 0,
+            },
+        },
+    }
+})
+
+export interface CheckoutDialogProps {
+    asset?: ReturnType<typeof useAsset>
+    open: boolean
+    onClose: () => void
+}
+
+export function CheckoutDialog(props: CheckoutDialogProps) {
+    const { asset, open, onClose } = props
+    const isVerified =
+        (!asset?.value?.isSoldOut &&
+            asset?.value?.totalSurplus > 0 &&
+            !asset?.value?.is24Auction &&
+            asset?.value?.priceInEth < 100000 &&
+            asset?.value?.trade?.isCanBuy) ??
+        false
+
+    const { t } = useI18N()
+    const { classes } = useStyles()
+
+    const chainId = useChainId()
+
+    const paymentTokens = resolvePaymentTokensOnCryptoartAI(chainId) ?? []
+    const selectedPaymentToken = first(paymentTokens)
+    const { token, balance } = useFungibleTokenWatched(selectedPaymentToken)
+
+    const [purchaseState, purchaseCallback, resetCallback] = usePurchaseCallback(
+        Number(asset?.value?.editionNumber),
+        asset?.value?.priceInWei,
+    )
+
+    const onCheckout = useCallback(() => {
+        purchaseCallback()
+    }, [purchaseCallback])
+
+    const { setDialog: setTransactionDialog } = useRemoteControlledDialog(
+        WalletMessages.events.transactionDialogUpdated,
+        (ev) => {
+            if (ev.open) return
+            resetCallback()
+        },
+    )
+
+    useEffect(() => {
+        if (purchaseState.type === TransactionStateType.UNKNOWN) return
+        setTransactionDialog({
+            open: true,
+            state: purchaseState,
+            summary: t('plugin_cryptoartai_buy') + ' ' + asset?.value?.title,
+        })
+    }, [purchaseState])
+
+    const validationMessage = useMemo(() => {
+        if (!isVerified) return t('plugin_collectible_check_tos_document')
+        if (asset?.value?.priceInEth > Number(formatBalance(balance.value, token?.value?.decimals, 6))) {
+            return t('plugin_collectible_insufficient_balance')
+        }
+        return ''
+    }, [isVerified, balance.value, asset?.value])
+
+    return (
+        <InjectedDialog title={t('plugin_cryptoartai_buy')} open={open} onClose={onClose}>
+            <DialogContent className={classes.content}>
+                <Card elevation={0}>
+                    <CardContent>
+                        <Box className={classes.container}>
+                            <Typography variant="body1" sx={{ marginBottom: 1 }}>
+                                <strong>
+                                    {t('plugin_cryptoartai_buy') +
+                                        ' ' +
+                                        asset?.value?.title +
+                                        ' ' +
+                                        asset?.value?.priceInEth +
+                                        ' ETH'}
```

Comment:
Yes, CryptoART.AI also support `Kovan` network. So you may post a link such as https://testweb.cryptoart.ai/gallery/detail/Notus%20Le/564 in twitter, and switch to `kovan` network, it will show the detail information

Reply:
why not use template literals build the string?

see http://es6-features.org/#StringInterpolation

Reply:
?